File: app/src/main/java/com/jiujiu/mvvmTemplate/di/ViewModelKey.java
package com.jiujiu.mvvmTemplate.di;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import androidx.lifecycle.ViewModel;
import dagger.MapKey;

@Retention(RetentionPolicy.RUNTIME)
@MapKey
public @interface ViewModelKey {
    Class<? extends ViewModel> value();
}

File: app/src/main/java/com/jiujiu/mvvmTemplate/di/scope/ViewModelKey.kt
package com.jiujiu.mvvmTemplate.di.scope

import androidx.lifecycle.ViewModel

import dagger.MapKey
import kotlin.reflect.KClass

@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@MapKey
annotation class ViewModelKey(val value: KClass<out ViewModel>)
